packages/nvidia_nat_core/src/nat/front_ends/fastapi/response_helpers.py

```
@@ -196,3 +199,74 @@ async def generate_streaming_response_full_as_str(payload: typing.Any,
+                                                  output_type: type | None = None) -> AsyncGenerator[str]:
+    """String-serialized variant of ``generate_streaming_response_atif``."""
+    try:
+        async for item in generate_streaming_response_atif(payload,
+                                                           session=session,
+                                                           streaming=streaming,
+                                                           result_type=result_type,
+                                                           output_type=output_type):
+            if isinstance(item, ResponseSerializable):
+                yield item.get_stream_data()
+            else:
+                raise ValueError("Unexpected item type in ATIF stream: " + str(type(item)))
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)